src/Doctrine/Odm/State/LinksHandlerTrait.php

```diff
@@ -37,7 +37,7 @@ private function handleLinks(Builder $aggregationBuilder, array $identifiers, ar
             return;
         }
 
-        $links = $this->getLinks($resourceClass, $operation, $context);
+        $links = $this->getLinks($context['resource_class'] ?? $resourceClass, $operation, $context);
```

Comment:
if `handleLinks` receives the correct class, we don't need to use the $context here also.